### PR TITLE
[SIMPLE_FORMS] feat: update form remediation logic to be configurable

### PR DESCRIPTION
**Pull request by pennja**
## Summary

This solution is designed to remediate form submissions which have failed submission and are over two weeks old. It is composed of several Ruby on Rails service objects which interact with each other.

The primary use-case for this solution is for form remediation. This process consists of the following:

1. Accept a form submission identifier.
1. Generate an archive payload consisting of the original form submission data as well as remediation specific documentation:
    1. Hydrate the original form submission
    1. Hydrate any original attachments that were a part of this submission
    1. Generate a JSON file with the original metadata from the submission
    1. Generate a manifest file to be used in the remediation process
1. Upload the generated archive as a .zip file onto the configured S3 bucket
1. Optionally return a presigned URL for accessing this file

This solution also provides a means for storing and retrieving a single .pdf copy of the originally submitted form.

The following image depicts how this solution is architected:

Error Remediation Architecture Diagram_2024-10-07_14-14-25

- This work updates the form remediation solution to be configurable and usable by all other va.gov teams.

## Related issue(s)

- [Form Submission Remediation - Genericize S3 bucket archiving solution for external team usage](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/issues/1954)

## Testing done

- [x] New logic is covered by unit tests

## Requested Feedback

Any

### CODE REVIEW

**Review by ndsprinkle — APPROVED** [review]

File: app/uploaders/simple_forms_api/form_remediation/uploader.rb
Comment (pennja) [inline app/uploaders/simple_forms_api/form_remediation/uploader.rb:1]: This file was renamed, it was originally `app/uploaders/veteran_facing_forms_remediation_uploader.rb`.

File: modules/simple_forms_api/app/services/simple_forms_api/form_remediation/jobs/archive_batch_processing_job.rb
Comment (pennja) [inline modules/simple_forms_api/app/services/simple_forms_api/form_remediation/jobs/archive_batch_processing_job.rb:1]: This file was renamed, it was originally [modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archive_builder.rb](https://github.com/department-of-veterans-affairs/vets-api/pull/18668/files#diff-d8f3527900e6c33b07dbf1f43c132cfdcf2c96ddca5c09ea9c11e8270a696f7c)

File: modules/simple_forms_api/app/services/simple_forms_api/form_remediation/s3_client.rb
Comment (pennja) [inline modules/simple_forms_api/app/services/simple_forms_api/form_remediation/s3_client.rb:1]: This file was renamed, it was originally: [modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archiver.rb](https://github.com/department-of-veterans-affairs/vets-api/pull/18668/files#diff-2fc329352337e96f831928c71b7d069b94c4b7514617e0275b360c09bcaf42af)

File: modules/simple_forms_api/app/services/simple_forms_api/form_remediation/submission_archive.rb
Comment (pennja) [inline modules/simple_forms_api/app/services/simple_forms_api/form_remediation/submission_archive.rb:1]: This file was renamed, it was originally: [modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archive_builder.rb](https://github.com/department-of-veterans-affairs/vets-api/pull/18668/files#diff-d8f3527900e6c33b07dbf1f43c132cfdcf2c96ddca5c09ea9c11e8270a696f7c)

File: modules/simple_forms_api/app/services/simple_forms_api/form_remediation/submission_remediation_data.rb
Comment (pennja) [inline modules/simple_forms_api/app/services/simple_forms_api/form_remediation/submission_remediation_data.rb:1]: This file was renamed, it was originally: [modules/simple_forms_api/app/services/simple_forms_api/s3/submission_builder.rb](https://github.com/department-of-veterans-affairs/vets-api/pull/18668/files#diff-46ee902e09732a2fced61ca7e17e4aef7a313a5027bb52efa2604162cccbd249)

File: modules/simple_forms_api/spec/services/form_remediation/jobs/archive_batch_processing_job_spec.rb
Comment (pennja) [inline modules/simple_forms_api/spec/services/form_remediation/jobs/archive_batch_processing_job_spec.rb:1]: This file was renamed, it was originally: [modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archive_handler.rb](https://github.com/department-of-veterans-affairs/vets-api/pull/18668/files#diff-63a6a9cc9b8a3658e029d7e9335cdf34d90e0d403e27d6baf09aaa347947b9e1)

File: modules/simple_forms_api/app/services/simple_forms_api/form_remediation/file_utilities.rb
Comment (pennja) [inline modules/simple_forms_api/app/services/simple_forms_api/form_remediation/file_utilities.rb:1]: This logic already existed within the archiver and archive builder and was pulled out into its own utility class.

File: lib/simple_forms_api/form_remediation/configuration/base.rb
Comment (pennja) [inline lib/simple_forms_api/form_remediation/configuration/base.rb:1]: The utility logic in this class was pulled out of: [modules/simple_forms_api/app/services/simple_forms_api/s3/utils.rb](https://github.com/department-of-veterans-affairs/vets-api/pull/18668/files#diff-768a1e3d39ac03296f271caf0b989e7fd1761755220ae52541269a840b6f9acb)

**Comment by LindseySaari** [comment]
[Per this conversation](https://dsva.slack.com/archives/CBU0KDSB1/p1728394510837969?thread_ts=1728329321.469959&cid=CBU0KDSB1), we are planning to move forward with the smaller PRs

**Pull request closed without merging**